src/PlayerOptions.cpp

```diff
@@ -1104,44 +1122,69 @@ bool PlayerOptions::FromOneModString( const RString &sOneMod, RString &sErrorOut
 	{
 	    if (sBit.find("x") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movex%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesX[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movex%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesX[i] )
+				break;
+				}
+			}
 	    }
 	    else if (sBit.find("y") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movey%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesY[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movey%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesY[i] )
+				break;
+				}
+			}
 	    }
 	    else if (sBit.find("z") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movez%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesZ[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movez%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesZ[i] )
+				break;
+				}
+			}
 	    }
 	}
 	else if( sBit == "zbuffer" )				m_bZBuffer = on;
 	else if( sBit == "cosecant" )				m_bCosecant = on;
 	else if( sBit == "visualdelay" )			m_fVisualDelay = level;
+	else if( level == 0 && disabledWindows.Compare(sBit)) // "No w1" etc.
```

Comment:
I think you can call `Compare` with a second argument to store the matches in the string. That way you don't have to split it yourself.

Reply:
`Compare` only gets the first match. The implementation here is basically copied from:
https://github.com/stepmania/stepmania/blob/984dc8668f1fedacf553f279a828acdebffc5625/src/smpackage/LanguagesDlg.cpp#L447

to get all the matches.